src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -48,13 +49,13 @@ public ViewBuffer(IViewBufferScope bufferScope, string name, int pageSize)
             _name = name;
             _pageSize = pageSize;
 
-            Pages = new List<ViewBufferPage>();
+            _pages = new ViewBufferPages();
         }
 
         /// <summary>
         /// Gets the backing buffer.
         /// </summary>
-        public IList<ViewBufferPage> Pages { get; }
+        public IReadOnlyList<ViewBufferPage> Pages => _pages;
```

Comment:
Better idea: I will add directly a `Count` property and a `operator[]` in the ViewBuffer. It will be useful in `ViewBuffer` implementation and in the test code.